src/AspNetCore/GraphQLHttpMiddleware.cs

```diff
@@ -54,7 +56,27 @@ private bool IsGraphQlRequest(HttpContext context)
 
         private async Task ExecuteAsync(HttpContext context, ISchema schema)
         {
-            var request = Deserialize<GraphQLQuery>(context.Request.Body);
+            // Handle requests as per recommendation at http://graphql.org/learn/serving-over-http/
+            var request = new GraphQLQuery();
+            if (context.Request.Method.Equals("GET", StringComparison.OrdinalIgnoreCase))
+            {
+                var qs = context.Request.Query;
+                request.Query = qs.TryGetValue("query", out StringValues queryValues) ? queryValues[0] : null;
+                request.Variables = qs.TryGetValue("variables", out StringValues variablesValues) ? JObject.Parse(variablesValues[0]) : null;
+                request.OperationName = qs.TryGetValue("operationName", out StringValues operationNameValues) ? operationNameValues[0] : null;
+            }
+            else if (context.Request.Method.Equals("POST", StringComparison.OrdinalIgnoreCase))
+            {
+                switch (context.Request.ContentType)
```

Comment:
Does this handle encodings as well?

Reply:
Not exactly sure what you mean by encodings. Could you give an example?